src/preact/bento-ce.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {applyStaticLayout} from '../layout';
+import {initLogConstructor} from '../log';
+import {toWin} from '../types';
+
+/** @type {typeof AMP.BaseElement} */
+let BaseElement;
+
+if (typeof AMP !== 'undefined' && AMP.BaseElement) {
+  BaseElement = AMP.BaseElement;
+} else {
+  class CeBaseElement {
+    /**
+     * @param {!Element} element
+     */
+    constructor(element) {
+      /** @const {!Element} */
+      this.element = element;
+
+      /** @const {!Window} */
+      this.win = toWin(element.ownerDocument.defaultView);
+    }
+
+    /**
+     * @param {typeof CeBaseElement} BaseElement
+     * @return {typeof HTMLElement}
+     */
+    static 'CustomElement'(BaseElement) {
+      initLogConstructor();
+      return class CustomElement extends HTMLElement {
+        /** */
+        constructor() {
+          super();
+
+          /** @const {!CeBaseElement} */
+          this.implementation = new BaseElement(this);
+        }
+
+        /** */
+        connectedCallback() {
+          applyStaticLayout(this);
+          this.implementation.buildCallback();
+          this.implementation.layoutCallback();
+        }
+      };
+    }
+
+    /**
+     * @param {function():undefined} cb
+     */
+    mutateElement(cb) {
+      Promise.resolve().then(cb);
```

Comment:
This would probably be best to be done via `setTimeout(0)`. That way the ordering of a mutation is still predictable. Vs a microtask ordering is completely random from the caller's point of view. Also, I think we might have more problems from overloading microtasks then tasks.

Reply:
just to confirm: while this fn is necessary while we still need `useResourcesNotify`, ideally we can rm this fn right?